packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/ClientProviders/ClientProviderTests.cs

```diff
@@ -465,24 +466,25 @@ public static IEnumerable<TestCaseData> BuildFieldsTestCases
                 }, false);
                 yield return new TestCaseData(new List<InputParameter>
                 {
+                    // have to explicitly set isRequired because we now call CreateParameter in buildFields
                     InputFactory.Parameter(
-                        "optionalParam",
+                        "optionalNullableParam",
                         InputPrimitiveType.String,
                         location: RequestLocation.None,
                         defaultValue: InputFactory.Constant.String("someValue"),
-                        kind: InputOperationParameterKind.Client),
+                        kind: InputOperationParameterKind.Client, isRequired: false),
                     InputFactory.Parameter(
                         "optionalParam2",
                         InputPrimitiveType.String,
                         location: RequestLocation.None,
                         defaultValue: InputFactory.Constant.String("someValue"),
-                        kind: InputOperationParameterKind.Client),
+                        kind: InputOperationParameterKind.Client, isRequired: true),
```

Comment:
similar here

Reply:
Also this answers the question in my [above comment](https://github.com/microsoft/typespec/pull/4392/files#r1794718226), despite the name of this parameter is **optional**Param2, actually it is required. This is very misleading. Could we align them?
And, changing the existing test case to represent a new scenario is not a good practice.
If we want to add some new scenarios, we should write a new test case.